framework/packages/abstract-sdk/src/apis/feegrant.rs

```diff
@@ -0,0 +1,348 @@
+    /// last period ended
+    pub period_reset: Option<Timestamp>,
+}
+
+impl PeriodicAllowance {
+    pub fn new(
+        basic: Option<BasicAllowance>,
+        period: Option<Duration>,
+        period_spend_limit: Vec<Coin>,
+        period_can_spend: Vec<Coin>,
+        period_reset: Option<Timestamp>,
+    ) -> Self {
+        Self {
+            basic,
+            period,
+            period_spend_limit,
+            period_can_spend,
+            period_reset,
+        }
+    }
+
+    pub fn to_proto(&self) -> feegrant::v1beta1::PeriodicAllowance {
+        feegrant::v1beta1::PeriodicAllowance {
+            basic: self.basic.clone().map(|b| b.to_proto()),
+            period: self.period.map(|p| prost_types::Duration {
+                seconds: p.as_secs() as i64,
+                nanos: 0,
+            }),
+            period_spend_limit: convert_coins(self.period_spend_limit.clone()),
+            period_can_spend: convert_coins(self.period_can_spend.clone()),
+            period_reset: self.period_reset.map(convert_stamp),
+        }
+    }
+}
+
+pub struct AllowedMsgAllowance {
+    /// allowance can be any of basic and periodic fee allowance.
+    pub allowance: Option<Box<dyn AllowedMsgAllowanceAllowance>>,
+    /// allowed_messages are the messages for which the grantee has the access.
+    pub allowed_messages: Vec<String>,
+}
+
+pub trait FeeGranterAllowance {
+    fn to_any(&self) -> Any;
+}
+
+impl FeeGranterAllowance for BasicAllowance {
+    fn to_any(&self) -> Any {
+        Any {
+            type_url: feegrant::v1beta1::BasicAllowance::TYPE_URL.to_string(),
+            value: self.clone().to_proto().encode_to_vec(),
+        }
+    }
+}
+
+impl FeeGranterAllowance for PeriodicAllowance {
+    fn to_any(&self) -> Any {
+        Any {
+            type_url: feegrant::v1beta1::PeriodicAllowance::TYPE_URL.to_string(),
+            value: feegrant::v1beta1::PeriodicAllowance {
+                basic: self.basic.clone().map(|b| b.to_proto()),
+                period: self.period.map(|p| prost_types::Duration {
+                    seconds: p.as_secs() as i64,
+                    nanos: 0,
+                }),
+                period_spend_limit: convert_coins(self.period_spend_limit.clone()),
+                period_can_spend: convert_coins(self.period_can_spend.clone()),
+                period_reset: self.period_reset.map(convert_stamp),
+            }
+            .encode_to_vec(),
+        }
+    }
+}
+
+impl FeeGranterAllowance for AllowedMsgAllowance {
+    fn to_any(&self) -> Any {
+        Any {
+            type_url: feegrant::v1beta1::AllowedMsgAllowance::TYPE_URL.to_string(),
+            value: feegrant::v1beta1::AllowedMsgAllowance {
+                allowance: self.allowance.as_ref().map(|a| a.to_any()),
+                allowed_messages: self.allowed_messages.clone(),
+            }
+            .encode_to_vec(),
+        }
+    }
+}
```

Comment:
We should probably use `Into<Any>`

Reply:
Unfortunately it's used in `dyn` trait object and `Into` is marked with `Sized`, which makes it not object-safe